src/sinks/nats.rs

```diff
@@ -186,32 +197,33 @@ impl StreamSink<Event> for NatsSink {
 
             let event_byte_size = event.size_of();
 
+            let finalizers = event.take_finalizers();
             let mut bytes = BytesMut::new();
-            if self.encoder.encode(event, &mut bytes).is_err() {
-                // Error is logged by `Encoder`.
-                continue;
-            }
+            self.encoder.encode(event, &mut bytes).map_err(|_| {
+                // Error is handled by `Encoder`.
+                finalizers.update_status(EventStatus::Errored);
+            })?;
```

Comment:
I think we want to continue the loop, no? I believe returning an error here will shutdown the sink.

Reply:
🤦 yeah, will fix that along with the integration tests

Reply:
Yes, I think you want to retain the old structure and just add the `update_status`.

Reply:
re-added in 902d5ee7c284281bc01b8c747bf4148f951c4338